src/dynamixel_control_hw.cpp

```diff
@@ -62,55 +71,94 @@ int main(int argc, char** argv)
-        dynamixel_corrections[k] = map_cor_i->second;
+    // Map for max speed (ID: max speed)
+    std::unordered_map<Protocol::id_t, double> dynamixel_max_speed;
+
+    // Retrieve the srevo parameter and fill maps above with its content.
+    std::unordered_map<Protocol::id_t, std::string> servos_map;
+    XmlRpc::XmlRpcValue servos_param; // temporary map, from parameter server
+    got_all_params &= nhParams.getParam("servos", servos_param);
+    ROS_ASSERT(servos_param.getType() == XmlRpc::XmlRpcValue::TypeStruct);
+    try {
+        for (XmlRpc::XmlRpcValue::ValueStruct::const_iterator it = servos_param.begin(); it != servos_param.end(); ++it) {
+            ROS_DEBUG_STREAM("Found servo: " << (std::string)(it->first));
+
+            Protocol::id_t id;
+            if (it->second.hasMember("id")) {
+                id = static_cast<int>(servos_param[it->first]["id"]);
+                ROS_DEBUG_STREAM("\tid: " << servos_param[it->first]["id"]);
+                dynamixel_map[id] = it->first;
+            }
+            else {
+                ROS_ERROR_STREAM("Actuator " << it->first
+                                             << " has no associated servo ID.");
+            }
+
+            if (it->second.hasMember("offset")) {
+                ROS_DEBUG_STREAM("\toffset: "
+                    << servos_param[it->first]["offset"]);
+                dynamixel_corrections[id]
+                    = static_cast<double>(servos_param[it->first]["offset"]);
+            }
+            if (it->second.hasMember("max_speed")) {
+                ROS_DEBUG_STREAM("\tmax_speed: "
+                    << servos_param[it->first]["max_speed"]);
+                dynamixel_max_speed[id]
+                    = static_cast<double>(servos_param[it->first]["max_speed"]);
+            }
+
+            if (it->second.hasMember("command_interface")) {
+                std::string mode_string
+                    = static_cast<std::string>(servos_param[it->first]["command_interface"]);
+                ROS_DEBUG_STREAM("\tcommand_interface: " << mode_string);
+
+                dynamixel_c_mode_map[id] = get_mode(mode_string, it->first);
+            }
+            else if (has_default_command_interface) {
+                ROS_DEBUG_STREAM("\tcommand_interface: "
+                    << default_command_interface << " (default)");
+                dynamixel_c_mode_map[id]
+                    = get_mode(default_command_interface, it->first);
+            }
+            else {
+                ROS_ERROR_STREAM("A command interface (speed or position) "
+                    << "should be declared for the actuator " << it->first
+                    << " or a default one should be defined with the parameter "
+                    << "'default_command_interface'.");
+            }
+        }
+    }
+    catch (XmlRpc::XmlRpcException& e) {
+        ROS_FATAL_STREAM("Exception raised when parsing the configuration: " << e.getMessage());
+        return 1;
```

Comment:
Most probably naive question, but just making sure. This should be enough to not allow incorrect configurations right? I.e., nothing is needed in the `dynamixel_hardware_interface.hpp` file. Just making sure..

Reply:
What is an incorrect configuration for you ?

The above code should only accept well-structures configuration. However, it does not prevent the addition of unused parameters (which are then ignored).
If a parameter is expected to be an integer but is of an other type, an exception is raised and causes the program to stop. Saddly, the exception message is not informative. I don't know how to improve this, except by splitting this try/catch in more, smaller ones.

Reply:
> The above code should only accept well-structures configuration.

That's what I correct configuration is for me..

> If a parameter is expected to be an integer but is of an other type, an exception is raised and causes the program to stop.

And that's what an incorrect configuration is for me. :smile: 

> However, it does not prevent the addition of unused parameters (which are then ignored).

This is ok.

> Saddly, the exception message is not informative. I don't know how to improve this, except by splitting this try/catch in more, smaller ones.

I think we should not bother too much for this.. It is not of that big importance..

Great! This is ok for me then!